middleware/kubernetes/setup.go

```diff
@@ -187,7 +189,38 @@ func kubernetesParse(c *caddy.Controller) (*Kubernetes, error) {
 						continue
 					}
 					return nil, fmt.Errorf("incorrect number of arguments for federation, got %v, expected 2", len(args))
+				case "autopath": // name zone
+					args := c.RemainingArgs()
+					k8s.AutoPathNdots = defautNdots
+					gotNdots := false
+					hostdomains := []string{}
+					for _, arg := range args {
+						if strings.HasPrefix(arg, ndotsOptPrefix) {
+							if gotNdots {
+								return nil, fmt.Errorf("found more than one ndots option, expected at most one")
+							}
+							ndots, err := strconv.Atoi(arg[len(ndotsOptPrefix):])
+							if err != nil {
+								return nil, fmt.Errorf("invalid ndots option for autopath, got '%v', expected an integer", ndots)
+							}
+							k8s.AutoPathNdots = ndots
+							gotNdots = true
+							continue
+						}
+						hostdomains = append(hostdomains, arg)
+					}
+					k8s.HostSearchPath = hostdomains
```

Comment:
ok - Less code is good.  To clarify , the ndots config would not come from the resolv.conf.  It should still be a separate parameter.